hazelcast/src/main/java/com/hazelcast/map/impl/operation/MergeOperation.java

```diff
@@ -59,16 +60,18 @@
     public MergeOperation() {
     }
 
-    MergeOperation(String name, List<SplitBrainMergeEntryView<Data, Data>> mergeEntries, SplitBrainMergePolicy policy) {
+    MergeOperation(String name, List<SplitBrainMergeEntryView<Data, Data>> mergeEntries, SplitBrainMergePolicy policy,
+                   boolean disableWanReplicationEvent) {
         super(name);
         this.mergeEntries = mergeEntries;
         this.mergePolicy = policy;
+        this.disableWanReplicationEvent = disableWanReplicationEvent;
     }
 
     @Override
     public void run() {
         hasMapListener = mapEventPublisher.hasEventListener(name);
-        hasWanReplication = mapContainer.getWanReplicationPublisher() != null && mapContainer.getWanMergePolicy() != null;
+        hasWanReplication = mapContainer.isWanReplicationEnabled() && !disableWanReplicationEvent;
```

Comment:
Thanks for optimising these checks, I always hated it when I saw them, knowing there was a method in `MapContainer`